session/session.go

```diff
@@ -160,9 +161,35 @@ func New(ctx context.Context, fatalErrCh chan error, logBackend *log.Backend, cf
 	s.setTimers(doc)
 
 	s.Go(s.worker)
+	s.tq = NewTimerQ(DelayQueue{nextQ: s.egressQueue, s: s, Lambda: doc.SendLambda, Max: doc.SendMaxInterval})
 	return s, nil
 }
 
+type DelayQueue struct {
+	Queue
+	nextQ EgressQueue
+	s *Session
+	Lambda float64
+	Max uint64
+}
+
+func (d DelayQueue) Push(e *Message) error {
+	wakeMsec := uint64(rand.Exp(rand.NewMath(), d.Lambda))
+	switch {
+	case wakeMsec > d.Max:
+		wakeMsec = d.Max
+	default:
+	}
+	d.s.log.Debugf("Re-enqueuing dropped message %x after %d ms", *e.ID, wakeMsec)
+	wakeInterval := time.Duration(wakeMsec) * time.Millisecond
+	go func() {
+		<-time.After(wakeInterval)
+		d.s.minclient.ForceFetch()
```

Comment:
wait... what? "ForceFetch" is a bad idea because this leaks timing information onto the network

Reply:
instead the client should only fetch when it's periodic timer fires

Reply:
ah, didn't realize this branch had some of these changes to session.go - I only meant to add timerq.go and timerq_test.go (but you can see how I was intending to use it). I think we need to think hard about the fetching strategy, because more often polling gives better latency but eats battery/bandwidth. Probably we want to tune the fetch rate to match the send rate.

Reply:
ok but you need to resolve this problem with your code and remove ForceFetch